src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveMetadata.cs

```diff
@@ -1,37 +1,46 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 namespace NuGet.Packaging.Signing
 {
     /// <summary>
-    /// This class is internally used to hold metadata about the signed package archive being verified.
+    /// This class is used to hold metadata about the signed package archive being verified.
     /// </summary>
-    internal sealed class SignedPackageArchiveMetadata
+    public class SignedPackageArchiveMetadata
     {
+
+        /// <summary>
+        /// List of central directory metadata ordered by the same order the central directory records are in the archive
+        /// </summary>
+        public List<CentralDirectoryMetadata> CentralDirectoryRecords { get; set; }
+
         /// <summary>
-        /// Position of the signature file header
+        /// Position in the archive (in bytes) where the file headers start relative to the start of the archive. Should typically be 0
         /// </summary>
-        internal long SignatureLocalFileHeaderPosition { get; set; }
+        public long StartOfFileHeaders { get; set; }
 
         /// <summary>
-        /// Total size of file entry for signature
-        /// This should include size of local file header + encryption header + file data + data descriptor
+        /// Position in the archive (in bytes) where the central directory starts relative to the start of the archive
         /// </summary>
-        internal long SignatureFileEntryTotalSize { get; set; }
+        public long StartOfCentralDirectory { get; set; }
+
 
         /// <summary>
-        /// Position of the signature file central directory header
+        /// Position in the archive (in bytes) where the central directory starts relative to the start of the archive
         /// </summary>
-        internal long SignatureCentralDirectoryHeaderPosition { get; set; }
+        public long EndOfCentralDirectory { get; set; }
 
         /// <summary>
-        /// Size of central directory header for signature file
+        /// Index of the signature central direcotry record.
+        /// If the CentralDirectories list is ordered by IndexInRecords this index indicates the position on the list for the signature.
```

Comment:
at the start it is ordered that way, but because this is not a readonly list it could be the case where this list is sorted by any order. For example when shifting entries it will have a random order.